### PR TITLE
Adding file constraint

### LINKED ISSUE #21: Support type for file uploads?
I'm wondering if there is a way (or something that could be added) to verify that a file is being uploaded?

### PR DESCRIPTION
**Pull request by bwillis**
Fixes #21

This will verify that the incoming parameter is a valid file upload type, either StringIO or File, as indicated in the [Rails guides](http://guides.rubyonrails.org/form_helpers.html#what-gets-uploaded).

### CODE REVIEW

**Comment by kintner** [comment]
:+1:  @bwillis I merged #28 can you rebase?

File: lib/stronger_parameters/constraints/file_constraint.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require 'stronger_parameters/constraints'
+
+module StrongerParameters
+  class FileConstraint < Constraint
+    def value(v)
+      if [File, StringIO].include? v.class
```
Comment (staugaard) [inline lib/stronger_parameters/constraints/file_constraint.rb]: `v.is_a?(File) || v.is_a?(StringIO)` would be better. Then you also catch subclasses.

File: lib/stronger_parameters/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module StrongerParameters
-  VERSION = '2.1.1'
+  VERSION = '2.1.2'
```
Comment (staugaard) [inline lib/stronger_parameters/version.rb]: don't bump the version in a PR

Comment (bwillis) [inline lib/stronger_parameters/version.rb]: I think my PR rebase got messed up, pulled in a bunch of other changes.

File: lib/stronger_parameters/constraints/file_constraint.rb (hunk 1 above)
Comment (grosser) [inline lib/stronger_parameters/constraints/file_constraint.rb]: should also work with `Rack::Test::UploadedFile` ... so maybe respond_to some methods ...

https://github.com/brynary/rack-test/blob/master/lib/rack/test/uploaded_file.rb

Comment (grosser) [inline lib/stronger_parameters/constraints/file_constraint.rb]: and ActionDispatch::Http::UploadedFile

https://github.com/rails/rails/blob/4e8880d29058ceb0549031c3f6fb95a88a26712e/actionpack/lib/action_dispatch/http/upload.rb#L68

Comment (bwillis) [inline lib/stronger_parameters/constraints/file_constraint.rb]: Thanks for the comments, updated, what do you think?

File: lib/stronger_parameters/version.rb (hunk 2 above)
Comment (bwillis) [inline lib/stronger_parameters/version.rb]: Fixed, tldr: last time I use the Github Desktop client.

**Comment by staugaard** [comment]
:+1:

File: README.md [hunk 3]
```diff
@@ -181,3 +181,4 @@ ActionController::Parameters.action_on_invalid_parameters = :log
 | Parameters.boolean             | Parameters.enum(true, false, 'true', 'false', 1, 0)                     |
 | Parameters.nil                 | value is nil                                                            |
 | Parameters.nil_string          | value is nil, '', 'undefined'                                           |
+| Parameters.file                | [File, StringIO].include? value.class                                   |
```
Comment (grosser) [inline README.md]: docs are outdated

File: lib/stronger_parameters/constraints/file_constraint.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+require 'stronger_parameters/constraints'
+
+module StrongerParameters
+  class FileConstraint < Constraint
+    VALID_FILE_TYPES = [
+      File,
+      StringIO,
+      Rack::Test::UploadedFile,
+      ActionDispatch::Http::UploadedFile
+    ].freeze
```
Comment (grosser) [inline lib/stronger_parameters/constraints/file_constraint.rb]: might be nice to not freeze that so users can add new classes easily

**Comment by bwillis** [comment]
Any more comments on this?

**Comment by staugaard** [comment]
Still :+1:

**Comment by grosser** [comment]
2.2.0 is out! :)

**Pull request merged**